src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp

```diff
@@ -554,6 +571,13 @@ void prepare_buffer_fusing::run(program& p) {
 
         program_helpers::do_for_types<reshape>(*node, [](reshape_node& node) {
             node.get_output_layout();
+
+            // Optimizing at prepare_buffer_fusing could propagate a padded input of an input nodes to Reshape.
+            // Reshape can be optimized out when only an outer axis(batch) has padding.
+            // For this case , it should re-calculate output padding size.
```

Comment:
Below code is only for oneDNN reshape users, If true please add comment and check logic onednn, 
Maybe the better place than reshape_inst.h

Reply:
I agree. It is moved into can_reshape_be_optimized.